hbase-client/src/main/java/org/apache/hadoop/hbase/client/MasterRegistry.java

```diff
@@ -259,17 +311,40 @@ private RegionLocations transformMetaRegionLocations(GetMetaRegionLocationsRespo
       .thenApply(GetClusterIdResponse::getClusterId);
   }
 
-  private ServerName transformServerName(GetActiveMasterResponse resp) {
-    return ProtobufUtil.toServerName(resp.getServerName());
+  private static boolean hasActiveMaster(GetMastersResponse resp) {
+    List<GetMastersResponseEntry> activeMasters =
+        resp.getMasterServersList().stream().filter(GetMastersResponseEntry::getIsActive).collect(
+        Collectors.toList());
+    return activeMasters.size() == 1;
+  }
+
+  private static ServerName filterActiveMaster(GetMastersResponse resp) {
+    List<GetMastersResponseEntry> activeMasters =
+        resp.getMasterServersList().stream().filter(GetMastersResponseEntry::getIsActive).collect(
+            Collectors.toList());
+    Preconditions.checkState(activeMasters.size() == 1);
```

Comment:
Don't think so. If you see getActiveMaster() call, "hasActiveMaster" is applied on the response. If that fails, we throw right away. This is only called on "thenApply()" which filters the output. Given the way the code is structured, I had to do it twice (since the call() takes a predicate that returns a boolean).

Reply:
But I do not think we will throw IllegalStateException? Usually we will throw IOException here? IllegalStateException is an unchecked exception...

Reply:
Its an unchecked exception because if this happens, it is a code bug. This means that someone called filterActiveMaster() without calling hasActiveMaster(). This is impossible to happen because its a private utility and we control the callers. It doesn't seem worth propagating the exceptions for it.